src/server.py

```diff
@@ -422,6 +416,25 @@ def run_create_jobtypes_query():
         query = "insert into uniquejobs (platform, buildtype, testtype) values ('%s', '%s', '%s')" % (j[0], j[1], j[2])
         cursor.execute(query)
 
+    # Add in preseed.json
+    preseed = []
+    with open(os.path.join(SCRIPT_DIR, 'preseed.json'), 'r') as fHandle:
+        preseed = json.load(fHandle)
+
+    for job in preseed:
+        found = False
+        for j in jobtypes:
+            if j[0] == job['platform'] and \
+               j[1] == job['buildtype'] and \
+               j[2] == job['name']:
+                found = True
+                break
+        if found:
+            continue
+
+        query = "insert into uniquejobs (platform, buildtype, testtype) values ('%s', '%s', '%s')" % (job['platform'], job['buildtype'], job['name'])
+        cursor.execute(query)
+
```

Comment:
This seems a bit wrong. Right now you have only talos jobs in preseed.json so you insert desktop jobs into uniquejobs before and then insert the talos jobs again here.

Instead insert talos jobs into jobtypes list directly and then insert into uniquejobs table together, that is a better non-confusing way.